server/src/main/java/org/elasticsearch/snapshots/AbortedSnapshotException.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.snapshots;
+
+public final class AbortedSnapshotException extends RuntimeException {
```

Comment:
Should this be an `IndexShardSnapshotException` so that it can be serialized across the wire?

Reply:
The reason I went for this (new exception type) was to explicitly not have this hit the wire ever. It's just here for flow control (not that that's great but that's how aborts work currently and as previously discussed it's not trivial to change that) and should never bubble up.